src/Post.php

```diff
@@ -994,23 +994,68 @@ public function category()
      *     {% endfor %}
      * {% endif %}
      * ```
-     * @param string|array $post_type _optional_ use to find children of a particular post type (attachment vs. page for example). You might want to restrict to certain types of children in case other stuff gets all mucked in there. You can use 'parent' to use the parent's post type or you can pass an array of post types.
+     * @param string|array $args _optional_ An array of arguments for the `get_children` function or a string/non-indexed array to use as the post type(s).
      * @return PostCollectionInterface
      */
-    public function children($post_type = 'any')
+    public function children($args = 'any')
     {
+        $post_type = 'any';
+        if (\is_string($args)) {
+            $post_type = $args;
+        } elseif (\array_values($args) === $args) {
+            $post_type = $args;
+        } elseif (\is_array($args)) {
+            $additional_args = $args;
+        }
+
```

Comment:
Could we maybe rewrite this using `wp_parse_args()`?

```suggestion
		if (\is_string($args) || \array_values($args) === $args) {
			$args = [
				'post_type' => $args,
			];
		}

		$args = wp_parse_args($args, [
            'post_parent' => $this->ID,
            'post_type' => 'any',
            'posts_per_page' => -1,
            'orderby' => 'menu_order title',
            'order' => 'ASC',
            'post_status' => 'publish',
        ]);
```

Then, we wouldn’t need `$additional_args`.

Reply:
Fixed this with some extra sauce to also get the parent status working